src/connection.cpp

```diff
@@ -41,272 +41,270 @@ namespace oxen::quic
         }
     }
 
-    int hook_func(
-            gnutls_session_t session, unsigned int htype, unsigned when, unsigned int incoming, const gnutls_datum_t* msg)
+    // ngtcp2 internal callback functions (that are deliberately source only, not a published
+    // header); we group them all in this struct to make them slightly easier to manage, but more
+    // importantly, because `Callbacks` is a friend-with-benefits of Endpoint that can touch its
+    // privates.
+    struct Callbacks
```

Comment:
I actually really like this idea for storing the callbacks we use and giving them access. I wonder if it would be useful in the future to have TestHelper extend that class so we could override and test specific callbacks, store other state, etc